va/va_cp.h

```diff
@@ -0,0 +1,420 @@
+ * the following conditions:
+ * 
+ * The above copyright notice and this permission notice (including the
+ * next paragraph) shall be included in all copies or substantial portions
+ * of the Software.
+ * 
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS
+ * OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF
+ * MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND NON-INFRINGEMENT.
+ * IN NO EVENT SHALL INTEL AND/OR ITS SUPPLIERS BE LIABLE FOR
+ * ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT,
+ * TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE
+ * SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+ */
+
+/**
+* \file va_cp.h
+* \brief The content protection API
+*
+* This file contains the \ref api_cp "Content Protection API".
+*/
+
+#ifndef VA_CP_H
+#define VA_CP_H
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+/**
+ * \defgroup api_cp Content Protection API
+ *
+ * @{
+ *
+ * The content protection API uses the same paradigm for codec.
+ * - Query for supported encryption attributes through VAConfigAttribEncryption.
+ * - create the specific configuration with the accepted encryption attributes got from the query.
+ * - Set up a codec pipeline;
+ 
+ * The basic operation steps for cenc decode are:
+ * - Query for supported encryption attributes through VAConfigAttribEncryption.
+ * - Create configuration with one of encryption attributes got from the above step.
+ * - Create a video decode.
+ * - Pass the render buffers with the corresponding data to CENC decode, including:
+ *   encrypted bitstream through VAProtectedSliceDataBufferType;
+ *   encryption paramters through VAEncryptionParameters.
+ * - Make the end of CENC call via vaEndCenc
+ * - Get CENC result via vaQueryCenc.
+ * - Pass the render buffers with the corresponding data to decode, including:
+ *   CENC status report number through VACencStatusParameterBufferType.
+ *   DPB information through VAPictureParameterBuffer.
+ * - Make the end of decode call via vaEndPicture
+ */
+
+/** attribute values for VAConfigAttribEncryption */
+#define VA_ENCRYPTION_TYPE_NONE             0x00000000
+#define VA_ENCRYPTION_TYPE_BASIC            0x00000001 
+#define VA_ENCRYPTION_TYPE_CENC_CBC         0x00000002
+#define VA_ENCRYPTION_TYPE_CENC_CTR_LENGTH  0x00000004
+#define VA_ENCRYPTION_TYPE_CENC_CTR         0x00000008
+#define VA_ENCRYPTION_TYPE_CTR_128          0x00000010
+
+/** \brief values for the encryption return status. */
+typedef enum
+{
+    /** \brief Indicate encryption operation is successful.*/ 
+    VA_ENCRYPTION_STATUS_SUCCESSFUL  = 0,
+    /** \brief Indicate encryption operation is incomplete.*/ 
+    VA_ENCRYPTION_STATUS_INCOMPLETE,
+    /** \brief Indicate encryption operation is error.*/ 
+    VA_ENCRYPTION_STATUS_ERROR,
+    /** \brief Indicate encryption result isn't available.*/ 
+    VA_ENCRYPTION_STATUS_UNAVAILABLE
```

Comment:
How do the three failure error codes differ?  It might help to suggest how the API user should respond to each one (retry, show error to end-user, ...).